src/test/java/org/jenkinsci/plugins/GithubAccessTokenPropertySEC797Test.java

```diff
@@ -242,17 +231,28 @@ private void onUserTeams(HttpServletRequest req, HttpServletResponse resp) throw
                 }});
             }
             
-            resp.getWriter().write(JSONArray.fromObject(responseBody).toString());
+            sendResponse(he, JSONArray.fromObject(responseBody).toString());
         }
         
-        private void onLoginOAuthAuthorize(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        private void onLoginOAuthAuthorize(HttpExchange he) throws IOException {
             String code = "test";
-            String state = req.getParameter("state");
-            resp.sendRedirect(jenkinsRule.getURL() + "securityRealm/finishLogin?code=" + code + "&state=" + state);
+            Fields fields = new Fields();
+            UrlEncoded.decodeUtf8To(he.getRequestURI().getQuery(), fields);
```

Comment:
Could use some copypasta like [this](https://stackoverflow.com/a/11733697) instead to completely avoid Jetty here, but this seems more maintainable and I doubt this method will change in the future (it is not tied to EE version). No strong preference here.